### PR TITLE
cocogitto 5.2.0 (new formula)

### PR DESCRIPTION
**Pull request by graelo**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/cocogitto.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    bash_complete.write Utils.safe_popen_read(bin/"cog", "generate-completions", "bash")
+    bash_completion.install bash_complete => "cog"
+
+    zsh_complete = buildpath/"zsh-completion"
+    zsh_complete.write Utils.safe_popen_read(bin/"cog", "generate-completions", "zsh")
+    zsh_completion.install zsh_complete => "_cog"
+
+    fish_complete = buildpath/"fish-completion"
+    fish_complete.write Utils.safe_popen_read(bin/"cog", "generate-completions", "fish")
+    fish_completion.install fish_complete => "cog.fish"
```
Comment (max-ae) [inline Formula/cocogitto.rb:23]:
```suggestion
    generate_completions_from_executable(bin/"cog", "generate-completions")
```

Comment (graelo) [reply Formula/cocogitto.rb]: Thanks!

File: Formula/cocogitto.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  url "https://github.com/cocogitto/cocogitto/archive/refs/tags/5.2.0.tar.gz"
+  sha256 "99f9dee05597d7721f6d046dbfefba5cb8d1c4ae22ced415f724affb3a6bd0cc"
+  license "MIT"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", "--locked", *std_cargo_args
+    generate_completions_from_executable(bin/"cog", "generate-completions",
+                                         shells: [:bash, :zsh, :fish], base_name: "cog")
```
Comment (max-ae) [inline Formula/cocogitto.rb:13]:
```suggestion
                                         base_name: "cog")
```
All three Shells is the default.

Comment (graelo) [reply Formula/cocogitto.rb:13]: sure, thanks

**Comment by graelo** [comment]
Thanks

**Comment by graelo** [comment]
Hi, is there anything I can do to move this further?

Thanks

**Comment by graelo** [comment]
I'll try rebasing on master

File: Formula/cocogitto.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  def install
+    system "cargo", "install", "--locked", *std_cargo_args
+    generate_completions_from_executable(bin/"cog", "generate-completions", base_name: "cog")
+  end
+
+  test do
+    assert_match "_cog()", shell_output("#{bin}/cog generate-completions bash")
+    assert_match "#compdef cog", shell_output("#{bin}/cog generate-completions zsh")
+    assert_match "set edit:completion:arg-completer[cog]", shell_output("#{bin}/cog generate-completions elvish")
+    assert_match "complete -c cog -n", shell_output("#{bin}/cog generate-completions fish")
```
Comment (SMillerDev) [inline Formula/cocogitto.rb:20]: Can you add a test that doesn't check just the completion output?

Comment (graelo) [reply Formula/cocogitto.rb:30]: Sure.

File: Formula/cocogitto.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    system "git", "config", "user.name", "'A U Thor'"
+    system "git", "config", "user.email", "author@example.com"
+    system "git", "commit", "-m", "chore: initial commit"
+    assert_equal "No errored commits", shell_output("#{bin}/cog check 2>&1").strip
+
+    # Check completions look correct
+    assert_match "_cog()", shell_output("#{bin}/cog generate-completions bash")
+    assert_match "#compdef cog", shell_output("#{bin}/cog generate-completions zsh")
+    assert_match "set edit:completion:arg-completer[cog]", shell_output("#{bin}/cog generate-completions elvish")
+    assert_match "complete -c cog -n", shell_output("#{bin}/cog generate-completions fish")
```
Comment (carlocab) [inline Formula/cocogitto.rb:30]:
```suggestion
```
I don't think we need this.

**Comment by graelo** [comment]
Kind bump ;)

**Comment by graelo** [comment]
Hello @carlocab, could you please have a look at my PR?

Kindly, thanks!

**Review by chenrui333 — DISMISSED** [review]

File: Formula/cocogitto.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  homepage "https://github.com/cocogitto/cocogitto"
+  url "https://github.com/cocogitto/cocogitto/archive/refs/tags/5.2.0.tar.gz"
+  sha256 "99f9dee05597d7721f6d046dbfefba5cb8d1c4ae22ced415f724affb3a6bd0cc"
+  license "MIT"
+
+  depends_on "rust" => :build
+  uses_from_macos "zlib"
+
+  def install
+    system "cargo", "install", "--locked", *std_cargo_args
```
Comment (carlocab) [inline Formula/cocogitto.rb:12]:
```suggestion
    system "cargo", "install", *std_cargo_args
```
`--locked` is part of [`std_cargo_args`](https://rubydoc.brew.sh/Formula.html#std_cargo_args-instance_method).

**Review by chenrui333 — APPROVED** [review]

**Comment by graelo** [comment]
Thanks @chenrui333 ❤️

**Comment by chenrui333** [comment]
@graelo, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**